server/server.go

```diff
@@ -335,67 +321,58 @@ func (s *server) Start(ctx context.Context, req *rpcpb.StartRequest) (*rpcpb.Sta
 		snapshotsDir:        s.cfg.SnapshotsDir,
 	})
 	if err != nil {
-		s.network = nil
 		return nil, err
 	}
 
-	if err := s.network.start(); err != nil {
+	s.log.Info("starting",
+		zap.String("exec-path", execPath),
+		zap.Uint32("num-nodes", numNodes),
+		zap.String("track-subnets", trackSubnets),
+		zap.Int32("pid", pid),
+		zap.String("root-data-dir", rootDataDir),
+		zap.String("plugin-dir", pluginDir),
+		zap.Any("chain-configs", req.ChainConfigs),
+		zap.String("global-node-config", globalNodeConfig),
+	)
+
+	if err := s.network.Start(); err != nil {
 		s.log.Warn("start failed to complete", zap.Error(err))
-		s.network = nil
+		s.stopAndRemoveNetwork()
 		return nil, err
 	}
 
-	// start non-blocking to install local cluster + custom chains (if applicable)
-	// the user is expected to poll cluster status
-	readyCh := make(chan struct{})
-	go s.network.startWait(ctx, chainSpecs, readyCh)
-
 	// update cluster info non-blocking
 	// the user is expected to poll this latest information
 	// to decide cluster/subnet readiness
 	go func() {
-		s.waitChAndUpdateClusterInfo("local cluster", readyCh, false)
-		s.waitChAndUpdateClusterInfo("custom chains", readyCh, true)
+		ctx, cancel := context.WithTimeout(context.Background(), waitForHealthyTimeout)
```

Comment:
previous semantics was to update info just after network start (before creating chains), and after creating chains. so, almost instantaneously the user can check cluster and node info, without waiting for longer chain creation operation for that

Reply:
We update `clusterInfo` before starting the network:

```go
	s.clusterInfo = &rpcpb.ClusterInfo{
		Pid:         pid,
		RootDataDir: rootDataDir,
	}
```

And also after:

```go
	go func() {
		ctx, cancel := context.WithTimeout(context.Background(), waitForHealthyTimeout)
		defer cancel()
		err := s.network.CreateChains(ctx, chainSpecs)
		s.mu.Lock()
		defer s.mu.Unlock()
		if err != nil {
			s.log.Error("network never became healthy", zap.Error(err))
			s.stopAndRemoveNetwork()
			return
		}
		s.updateClusterInfo()
		s.log.Info("network healthy")
	}()
```

Reply:
the previous situation was for clusterInfo to be updated after starting the network, with the node names etc, and secondly after creating the blockchains, with the subnet/blockchain infoi

Reply:
anyway we can keep this, it is a minor semantic change not noticeable for users